docs/knn/jni-library.md

```diff
@@ -0,0 +1,13 @@
+---
+layout: default
+title: JNI Library
+nav_order: 5
+parent: k-NN
+has_children: false
+---
+
+# JNI Library
+In order to integrate nmslib's approximate k-NN functionality, which is implemented in C++, into the k-NN plugin, which is implemented in Java, we created a Java Native Interface library. Check out [this wiki](https://en.wikipedia.org/wiki/Java_Native_Interface) to learn more about JNI. This library allows the k-NN plugin to leverage nmslib's functionality.
+
+## Artifacts
```

Comment:
I think this section is my only real beef with this PR. It feels very much like "if you're going to build this plugin yourself, here's some information you need to know," which is better-suited to the README. I think I'd prefer a single sentence in the previous paragraph that just says:

```
For more information about how we build the JNI library binary and how to get the most of it in your production environment, see [here](https://github.com/opendistro-for-elasticsearch/k-NN#jni-library-artifacts).
```

Reply:
That makes sense. I will update